### ISSUE TITLE
Publish command does not work and authentication fails 

### ISSUE THREAD

**Issue by msh855** [quarto-dev/quarto-cli#2994]
### Bug description

Hi, 

I am trying to create a blog deployed by GitHub Pages. However, the process [you are describing](https://quarto.org/docs/publishing/github-pages.html) is failing when I try to run the `quarto publish` command 

Here is what I get in my terminal: 

![image](https://user-images.githubusercontent.com/114295360/197402763-f0727139-8e1a-4965-9fdf-32c6dcf9a9f5.png)

My repo is hosted in this link: https://github.com/msh855/TheQuantEconomist 

As a side note, the way I created the `ph-pages` branch is by just creating a new branch from the `main` branch as the documentation on your website was not very clear on how to create `gh-page` branch. 

I am using Mac, if this helps.

**Comment by cscheid** [comment]
It's probable that the reason the process is failing is that there's an existing gh-pages branch. You shouldn't need to create that branch yourself. Can you try to run the process on a different repository without creating the branch?

**Comment by msh855** [comment]
No, nothing to do with this. I tried before I create gh-pqges and also fails. 

Please read the error message carefully. It is an authentication error that I am not sure how to correct

**Comment by dragonstyle** [comment]
https://stackoverflow.com/questions/68781928/support-for-password-authentication-was-removed-on-august-13-2021

**Comment by msh855** [comment]
Thanks, seem that it worked.

I think the documentation needs to be updated. 

Also, it needs to make more clear how the `gh-pages` is created and at which stage of the process. Is quarto creating the `gh-pages` branch via the `quarto publish gh-pages` command or does the user need to create a new branch and name it `gh-pages` before publishing?

It is not very clear this part and as the reader goes through the instructions it suddenly sees out of the blue "change the branch to gh-page"

**Comment by cwickham** [comment]
Instructions for creating a branch were added in https://github.com/quarto-dev/quarto-web/commit/09c9f796d76121ba90fcefdab6452c9ea8fb0a27
